sw/device/tests/aes_idle.c

```diff
@@ -0,0 +1,229 @@
+        kDifClkmgrOk);
+  CHECK(enabled == expected);
+}
+
+void test_aes_active(const dif_clkmgr_t *clkmgr) {
+  // Verify that the AES clk hint status within clkmgr now reads 1 (AES is not idle),
+  //   before the AES operation is complete.
+  bool enabled;
+  CHECK(dif_clkmgr_hintable_clock_get_enabled(clkmgr, kTopEarlgreyHintableClocksMainAes, &enabled) ==
+        kDifClkmgrOk);
+  CHECK(enabled == true);
+}
+
+void test_aes_idle(const dif_clkmgr_t *clkmgr) {
+  // Verify that the AES clk hint status within clkmgr now reads 0 again (AES is idle),
+  //   after the AES operation is complete.
+  bool enabled;
+  CHECK(dif_clkmgr_hintable_clock_get_enabled(clkmgr, kTopEarlgreyHintableClocksMainAes, &enabled) ==
+        kDifClkmgrOk);
+  CHECK(enabled == false);
+}
+
+
+/**
+ * Test that AES 'hintable' clock, indirectly controlled by software,
+ * can have its hint toggled and status checked.
+ */
+void test_aes_clock(const dif_clkmgr_t *clkmgr) {
+
+  // Get the initial state of the hint for the clock The clock hint might be
+  // enabled or disabled depending on reset behavior - either is fine for the
+  // purposes of this test.
+  bool enabled;
+  CHECK(dif_clkmgr_hintable_clock_get_hint(clkmgr, kTopEarlgreyHintableClocksMainAes, &enabled) ==
+        kDifClkmgrOk);
+
+  // Toggle the hint twice so that it ends up in its original state.
+  for (int j = 0; j < 2; ++j) {
+    bool expected = !enabled;
+    CHECK(dif_clkmgr_hintable_clock_set_hint(
+              clkmgr, kTopEarlgreyHintableClocksMainAes,
+              expected ? kDifClkmgrToggleEnabled
+                       : kDifClkmgrToggleDisabled) == kDifClkmgrOk);
+    CHECK(dif_clkmgr_hintable_clock_get_hint(clkmgr, kTopEarlgreyHintableClocksMainAes, &enabled) ==
+          kDifClkmgrOk);
+    CHECK(enabled == expected);
+
+    // If the clock hint is enabled then the clock should always be enabled.
+    if (enabled) {
+      bool status = false;
+      CHECK(dif_clkmgr_hintable_clock_get_enabled(clkmgr, kTopEarlgreyHintableClocksMainAes, &status) ==
+            kDifClkmgrOk);
+      CHECK(status, "clock %u hint is enabled but status is disabled", kTopEarlgreyHintableClocksMainAes);
+    }
+  }
+}
+
+
+bool test_main(void) {
+  const dif_clkmgr_params_t params = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR),
+      .last_gateable_clock = kTopEarlgreyGateableClocksLast,
+      .last_hintable_clock = kTopEarlgreyHintableClocksLast,
+  };
+
+  dif_aes_t aes;
+  dif_clkmgr_t clkmgr;
+
+  CHECK(dif_clkmgr_init(params, &clkmgr) == kDifClkmgrOk);
+
+  LOG_INFO("Entropy complex ini...");
+  // First of all, we need to get the entropy complex up and running.
+  entropy_testutils_boot_mode_init();
+
+  LOG_INFO("AES ini...");
+  // Initialise AES.
+  CHECK_DIF_OK(
+      dif_aes_init(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR), &aes));
+  CHECK_DIF_OK(dif_aes_reset(&aes));
+
+  LOG_INFO("Set AES hint");
+  set_aes_hint(&clkmgr);
```

Comment:
Readhist status, verify it reads back 0. Set hint back to 1 so that SW can program AES.

Reply:
I can write a test that gates and ungates the clock a random number of times in between the calls to the AES dif.